### ISSUE TITLE
Feature Request: Abort on request

### ISSUE THREAD

**Issue by melroy89** [HelgeSverre/ollama-gui#14]
As a user I want to be able to abort my request/message. For example give me a "stop" button after I sent my message (but not yet received a response back). Allowing to stop ollama to fully abort the request and not consume any resources anymore, until I sent again a (new) message.

**Comment by HelgeSverre** [comment]
Hmm, is this 'as simple' as using cancellation tokens with fetch, or am i underedtimating this?

**Comment by melroy89** [comment]
Not sure to be honest, since I didn't use the Ollama API directly: https://github.com/jmorganca/ollama/blob/main/docs/api.md#request

I really hope the API allows you to cancel the request again. It's not just the fetch it's about aborting the generate part in ollama as well. Similar to hitting 'ctrl+c' when within the ollama CLI tool (during a `ollama run <...>`), which will stop the generate part and give me back the prompt (it will also stop using resources).

**Comment by HelgeSverre** [comment]
Fixed.

This is available on the main branch now, `git pull` to update your local version, or use the [hosted version](https://ollama-gui.vercel.app/), which has also been updated.

**Comment by melroy89** [comment]
Great! Thanks.

**Comment by melroy89** [comment]
@HelgeSverre I tested it today. But it actually does not work as expected. It still consumes resources after I pressed the 'Stop' button.

![image](https://github.com/HelgeSverre/ollama-gui/assets/628926/ed62f058-6b0d-4cdd-8a7a-773049c278f9)

While I do indeed do not get a response back, in the background it's still using the computer resources. You maybe have a fast PC with a dedicated GPU with Tensor Cores, I do not.
